tests/expressions_case/tests.py

```diff
@@ -0,0 +1,1136 @@
+                    When(integer2=F('integer'), then=Value(2)),
+                    When(integer2=F('f_plus_1'), then=Value(3)),
+                    output_field=models.IntegerField()
+                )
+            ).order_by('pk'),
+            [(3, 4), (2, 2), (3, 4)],
+            transform=attrgetter('integer', 'integer2')
+        )
+
+    def test_filter_with_annotation_in_predicate(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.annotate(
+                f_plus_1=F('integer') + 1
+            ).filter(
+                integer2=Case(
+                    When(f_plus_1=3, then=Value(3)),
+                    When(f_plus_1=4, then=Value(4)),
+                    default=Value(1),
+                    output_field=models.IntegerField()
+                )
+            ).order_by('pk'),
+            [(1, 1), (2, 3), (3, 4), (3, 4)],
+            transform=attrgetter('integer', 'integer2')
+        )
+
+    def test_filter_with_aggregation_in_value(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.values(*self.non_lob_fields).annotate(
+                min=Min('fk_rel__integer'),
+                max=Max('fk_rel__integer')
+            ).filter(
+                integer2=Case(
+                    When(integer=2, then='min'),
+                    When(integer=3, then='max')
+                )
+            ).order_by('pk'),
+            [(3, 4, 3, 4), (2, 2, 2, 3), (3, 4, 3, 4)],
+            transform=itemgetter('integer', 'integer2', 'min', 'max')
+        )
+
+    def test_filter_with_aggregation_in_condition(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.values(*self.non_lob_fields).annotate(
+                min=Min('fk_rel__integer'),
+                max=Max('fk_rel__integer')
+            ).filter(
+                integer=Case(
+                    When(integer2=F('min'), then=Value(2)),
+                    When(integer2=F('max'), then=Value(3))
+                )
+            ).order_by('pk'),
+            [(3, 4, 3, 4), (2, 2, 2, 3), (3, 4, 3, 4)],
+            transform=itemgetter('integer', 'integer2', 'min', 'max')
+        )
+
+    def test_filter_with_aggregation_in_predicate(self):
+        self.assertQuerysetEqual(
+            CaseTestModel.objects.values(*self.non_lob_fields).annotate(
+                max=Max('fk_rel__integer')
+            ).filter(
+                integer=Case(
+                    When(max=3, then=Value(2)),
+                    When(max=4, then=Value(3))
+                )
+            ).order_by('pk'),
+            [(2, 3, 3), (3, 4, 4), (2, 2, 3), (3, 4, 4), (3, 3, 4)],
+            transform=itemgetter('integer', 'integer2', 'max')
+        )
+
+    def test_update(self):
+        CaseTestModel.objects.update(
+            string=Case(
+                When(integer=1, then=Value('one')),
+                When(integer=2, then=Value('two')),
+                default=Value('other')
+            )
+        )
+
```

Comment:
since the brackets on their own lines add whitespace, I think you could omit the blank line between queries/assertions in most cases to make this file a bit shorter.